bitnami/common/templates/_validations.tpl

```diff
@@ -56,39 +56,109 @@ Validate value params:
 {{- end -}}
 
 {{/*
-Validate a mariadb required password must not be empty.
+Validate MariaDB required passwords are not empty.
 
 Usage:
-{{ include "common.validations.values.mariadb.passwords" (dict "secret" "secretName" "context" $) }}
-
-Validate value params:
-  - secret - String - Required. Name of the secret where mysql values are stored, e.g: "mysql-passwords-secret"
+{{ include "common.validations.values.mariadb.passwords" (dict "secret" "secretName" "subchart" false "context" $) }}
+Params:
+  - secret - String - Required. Name of the secret where MariaDB values are stored, e.g: "mysql-passwords-secret"
+  - subchart - Boolean - Optional. Whether MariaDB is used as subchart or not. Default: false
 */}}
 {{- define "common.validations.values.mariadb.passwords" -}}
-  {{- if and (not .context.Values.mariadb.existingSecret) .context.Values.mariadb.enabled -}}
-    {{- $requiredPasswords := list -}}
+  {{- $existingSecret := include "common.mariadb.values.existingSecret" . -}}
+  {{- $enabled := include "common.mariadb.values.enabled" . -}}
+  {{- $architecture := include "common.mariadb.values.architecture" . -}}
+  {{- $valueKeyRootPassword := printf "%s.rootPassword" (include "common.mariadb.values.key.auth" .) -}}
+  {{- $valueKeyUsername := printf "%s.username" (include "common.mariadb.values.key.auth" .) . -}}
+  {{- $valueKeyPassword := printf "%s.password" (include "common.mariadb.values.key.auth" .) . -}}
+  {{- $valueKeyReplicationPassword := printf "%s.replicationPassword" (include "common.mariadb.values.key.auth" .) -}}
 
-    {{- if .context.Values.mariadb.secret.requirePasswords -}}
-      {{- $requiredRootMariadbPassword := dict "valueKey" "mariadb.rootUser.password" "secret" .secretName "field" "mariadb-root-password" -}}
-      {{- $requiredPasswords = append $requiredPasswords $requiredRootMariadbPassword -}}
+  {{- if and (not $existingSecret) (eq $enabled "true") -}}
+    {{- $requiredPasswords := list -}}
 
-      {{- if not (empty .context.Values.mariadb.db.user) -}}
-          {{- $requiredMariadbPassword := dict "valueKey" "mariadb.db.password" "secret" .secretName "field" "mariadb-password" -}}
-          {{- $requiredPasswords = append $requiredPasswords $requiredMariadbPassword -}}
-      {{- end -}}
+    {{- $requiredRootPassword := dict "valueKey" $valueKeyRootPassword "secret" .secret "field" "mariadb-root-password" -}}
+    {{- $requiredPasswords = append $requiredPasswords $requiredRootPassword -}}
 
-      {{- if .context.Values.mariadb.replication.enabled -}}
-          {{- $requiredReplicationPassword := dict "valueKey" "mariadb.replication.password" "secret" .secretName "field" "mariadb-replication-password" -}}
-          {{- $requiredPasswords = append $requiredPasswords $requiredReplicationPassword -}}
-      {{- end -}}
+    {{- $valueUsername := index $.context.Values $valueKeyUsername }}
```

Comment:
did you test`index` when it is a long path `path.to.the.value`? I don't remember well but I think it was not working fine for me.